### PR TITLE
chore(config): Move `assume_role` option to `auth.assume_role`

### LINKED ISSUE #4200: Move root level `assume_role` option to `auth.assume_role`
Cleanup for #4199.

### PR DESCRIPTION
**Pull request by ktff**
Closes #4200

### Open questions

- [x] This also moves `assume_role` in source `aws_s3` and sink `aws_sqs`. @binarylogic this is fine?

### Notes
`AWSAuthentication` is an `enum` since #4199 will add an additional variant.

### CODE REVIEW

**Comment by binarylogic** [comment]
> This also moves assume_role in source aws_s3 and sink aws_sqs. @binarylogic this is fine?

Yes, that is preferred.

File: src/rusoto/auth.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            r#"
+            auth.assume_role = "root"
+        "#,
+        )
+        .unwrap();
+    }
+
+    #[test]
+    fn parsing_both_assume_role() {
+        toml::from_str::<ComponentConfig>(
```
Comment (jszwedko) [inline src/rusoto/auth.rs:84]: Should we assert that it chooses `auth.assume_role` here? We could add assertions to the others as well. I see we'd really just be asserting that serde does what it should but we've been surprised by it before. This one is actually logic that we have though.

**Review by jszwedko — APPROVED** [review]
This looks good to me! Just one comment about test assertions.

File: src/rusoto/auth.rs (hunk 1 above)
Comment (ktff) [reply src/rusoto/auth.rs:84]: One other test will fail in that case, but yea, better be sure and also test that here.

**Pull request merged**